src/NetscriptFunctions/StockMarket.ts

```diff
@@ -388,5 +388,47 @@ export function NetscriptStockMarket(player: IPlayer, workerScript: WorkerScript
       workerScript.log("stock.purchase4SMarketDataTixApi", () => "Purchased 4S Market Data TIX API");
       return true;
     },
+    purchaseWseAccount: function (): boolean {
+      helper.updateDynamicRam("PurchaseWseAccount", getRamCost(player, "stock", "purchaseWseAccount"));
+
+      if (player.hasWseAccount) {
+        workerScript.log("stock.purchaseWseAccount", () => "Already purchased WSE Account");
+        return true;
+      }
+
+      if (player.money < getStockMarketWseCost()) {
+        workerScript.log(
+          "stock.purchaseWseAccount",
+          () => "Not enough money to purchase WSE Account Access",
+        );
+        return false;
+      }
+
+      player.hasWseAccount = true;
+      player.loseMoney(getStockMarketWseCost(), "stock");
+      workerScript.log("stock.purchaseWseAccount", () => "Purchased WSE Account Access");
+      return true;
+    },
+    purchaseTixApi: function (): boolean {
+      helper.updateDynamicRam("PurchaseTixApi", getRamCost(player, "stock", "purchaseTixApi"));
```

Comment:
should be `stock.purchaseTixAPI` and you're missing the corresponding line in RamCostGenerator.tsx

dito 392